tests/rptest/tests/node_metrics_test.py

```diff
@@ -0,0 +1,132 @@
+
+    def _node_disk_total_bytes(self) -> List[int]:
+        return self._get_metrics_vals("storage_disk_total_bytes")
+
+    def _node_disk_free_bytes(self) -> List[int]:
+        return self._get_metrics_vals("storage_disk_free_bytes")
+
+    def _node_disk_space_alert(self) -> List[int]:
+        return self._get_metrics_vals("storage_disk_free_space_alert")
+
+    def _count_greater(self, l1: List[T], l2: List[T]) -> int:
+        """ return number of elements in l1 that were *strictly greater* than their
+        counterpart in l2. """
+        num_greater = 0
+        for i, v in enumerate(l1):
+            if v > l2[i]:
+                num_greater += 1
+            self.redpanda.logger.debug(
+                f"count_greater({v} - {l2[i]} = {v - l2[i]}) -> {num_greater}")
+
+        self.redpanda.logger.debug(
+            f"count_greater: {l1} / {l2} -> {num_greater}")
+        return num_greater
+
+    def _produce_consumed_space(self, orig_free: List[int]) -> bool:
+        """ Test helper: produce about 10MiB of data and return true if any of
+        the nodes saw a reduction in free space.
+        """
+        num_records = 10240
+        record_size = 1024
+
+        # Produce data and confirm metrics update
+        ktools = KafkaCliTools(self.redpanda)
+        ktools.produce(self.topic, num_records, record_size, acks=-1)
+
+        new_free = self._node_disk_free_bytes()
+        return self._count_greater(orig_free, new_free) > 0
+
+    @cluster(num_nodes=3)
+    def test_node_storage_metrics(self):
+
+        # disk metrics are updated via health monitor's periodic tick().
+        t0 = time()
+        wait_until(
+            lambda: all_greater_than_zero(self._node_disk_total_bytes()),
+            timeout_sec=15,
+            err_msg="Disk metrics not populated before timeout.")
+        t1 = time()
+
+        # check that metrics exist and remember the values
+        orig_total = self._node_disk_total_bytes()
+        orig_free = self._node_disk_free_bytes()
+
+        self.redpanda.logger.debug(
+            f'orig total {orig_total}, orig free {orig_free}')
+
+        # for alert field, just confirm it exists and in valid range. Actual
+        # logic is covered by local_monitor_test.cc
+        alerts = self._node_disk_space_alert()
+        assert len(alerts) == 3  # for each node
+        assert all([x >= 0 for x in alerts])
+
+        # We want to use up storage and verify that our free space decreases.
+        # This is currently complicated by our docker test setup; running it
+        # locally, all containers report space free on my primary laptop drive
+        # (which they see via overlayfs).  When running, my tests see varying
+        # free space caused by other processes on the system. This required a
+        # long retry loop below to eventually use up that space, about 5% of the
+        # time. We should look using separate per-container volumes, at least
```

Comment:
can you use a small ramdisk or tmpfs in the test? assuming that works in the docker environment then it should be a small change to redpanda.py to teach it a new path? then the test can be very fast because it won't have any interference.

Reply:
I was thinking something similar.. Didn't do this yet because I still need better testing strategy for full disk, and might end up replacing this with a FUSE shim or RP-based filesystem filter/injector.

Reply:
got it makes sense